primitives/state-machine/src/lib.rs

```diff
@@ -297,6 +301,9 @@ impl<'a, B, H, N, Exec> StateMachine<'a, B, H, N, Exec> where
 			None => &mut cache,
 		};
 
+		self.overlay.enter_runtime()
+			.expect("This function is supposed to be called from the client only.");
```

Comment:
For that reason you do not see a qed at the end. I am not buying that `expect` is only for proofs. Look at all the `expects` in the substrate codebase. Those are all just `unwrap` with a message. 

For example all the `expect` sp_io:
https://github.com/paritytech/substrate/blob/d735e4d0b5378c227f81a5127a1d4544de112fd8/primitives/io/src/lib.rs#L151

Or in the same crate just running expect with a blanket:
https://github.com/paritytech/substrate/blob/d735e4d0b5378c227f81a5127a1d4544de112fd8/primitives/state-machine/src/ext.rs#L39

That said, how would you handle this invariant there, instead?

Reply:
The expects you found are in runtime related code, which is allowed to fail/panic. This will not bring down the full node^^ 

> "StateMachine is never called from the runtime; qed."

Sounds good.

Reply:
I see. Changed the wording.